### DISCUSSION TITLE
Override combobox enter key.

### DISCUSSION THREAD

**Discussion by Quetzalcoatl2435** [AvaloniaUI/Avalonia#16088] [Q&A] [answered]
The default enter key behaviour for combobox in avalonia 11 is set to open the dropdown list. How can I override this to do something else?

**Comment by workgroupengineering** [comment, marked as answer]
```c#
public class MyComboBox:ComboBox
{
 
  protected override Type StyleKeyOverride => typeof(ComboBox); // for reuse the style

  protected override void OnKeyDown(KeyEventArgs e)
  {
      // Your logic
  }
}
``` 

Ref.

https://github.com/AvaloniaUI/Avalonia/blob/143399f65abc8b97b479b1f8ed9d74b1a6a65303/src/Avalonia.Controls/ComboBox.cs#L214-L266

**Reply by maxkatz6** [reply]
You also can handle KeyDown event with Tunnel strategy.

**Reply by Quetzalcoatl2435** [reply]
> ```cs
> public class MyComboBox:ComboBox
> {
>  
>   protected override Type StyleKeyOverride => typeof(ComboBox); // for reuse the style
> 
>   protected override void OnKeyDown(KeyEventArgs e)
>   {
>       // Your logic
>   }
> }
> ```
> 
> Ref.
> 
> https://github.com/AvaloniaUI/Avalonia/blob/143399f65abc8b97b479b1f8ed9d74b1a6a65303/src/Avalonia.Controls/ComboBox.cs#L214-L266

Custom controls went completely over my head. My mind must have blanked out for a moment there. :D

```
public class CustomComboBox : ComboBox
{
    protected override Type StyleKeyOverride => typeof(ComboBox);
    protected override void OnKeyDown(KeyEventArgs e)
    {
        if (IsDropDownOpen || e.Key != Key.Enter)
            base.OnKeyDown(e);
    }
}
```

That did it for me. Thanks for the suggestion.
